types/lib/legacy/polymer.dom.d.ts

```diff
@@ -0,0 +1,145 @@
+
+    /**
+     * Disconnects an observer previously created via `observeNodes`
+     *
+     * @param observerHandle Observer instance
+     *   to disconnect.
+     */
+    unobserveNodes(observerHandle: Polymer.FlattenedNodesObserver|null): void|null;
+
+    /**
+     * Provided as a backwards-compatible API only.  This method does nothing.
+     */
+    notifyObserver(): void|null;
+
+    /**
+     * Returns true if the provided node is contained with this element's
+     * light-DOM children or shadow root, including any nested shadow roots
+     * of children therein.
+     *
+     * @param node Node to test
+     * @returns Returns true if the given `node` is contained within
+     *   this element's light or shadow DOM.
+     */
+    deepContains(node: Node|null): boolean;
+
+    /**
+     * Returns the root node of this node.  Equivalent to `getRoodNode()`.
+     *
+     * @returns Top most element in the dom tree in which the node
+     * exists. If the node is connected to a document this is either a
+     * shadowRoot or the document; otherwise, it may be the node
+     * itself or a node or document fragment containing it.
+     */
+    getOwnerRoot(): Node|null;
+
+    /**
+     * For slot elements, returns the nodes assigned to the slot; otherwise
+     * an empty array. It is equivalent to `<slot>.addignedNodes({flatten:true})`.
+     *
+     * @returns Array of assigned nodes
+     */
+    getDistributedNodes(): Array<Node|null>|null;
+
+    /**
+     * Returns an array of all slots this element was distributed to.
+     *
+     * @returns Description
+     */
+    getDestinationInsertionPoints(): Array<HTMLSlotElement|null>|null;
+
+    /**
+     * Calls `importNode` on the `ownerDocument` for this node.
+     *
+     * @param node Node to import
+     * @param deep True if the node should be cloned deeply during
+     *   import
+     * @returns Clone of given node imported to this owner document
+     */
+    importNode(node: Node|null, deep: boolean): Node|null;
+
+    /**
+     * @returns Returns a flattened list of all child nodes and nodes assigned
+     * to child slots.
+     */
+    getEffectiveChildNodes(): any[]|null;
+
+    /**
+     * Returns a filtered list of flattened child elements for this element based
+     * on the given selector.
+     *
+     * @param selector Selector to filter nodes against
+     * @returns List of flattened child elements
+     */
+    queryDistributedElements(selector: string): Array<HTMLElement|null>|null;
+  }
+
+
+  /**
+   * Legacy DOM and Event manipulation API wrapper factory used to abstract
+   * differences between native Shadow DOM and "Shady DOM" when polyfilling on
+   * older browsers.
+   *
+   * Note that in Polymer 2.x use of `Polymer.dom` is no longer required and
+   * in the majority of cases simply facades directly to the standard native
+   * API.
+   */
+  function dom(obj: Node|Event|null): DomApi|EventApi|null;
```

Comment:
I would specify this as the following:

```ts
function dom(obj: Node): DomApi;
function dom(obj: Event): EventApi;
function dom(obj: null): DomApi;
```

Reply:
That would be really nice, but no way to represent this with Closure so we'd need to manually override this. Don't have that mechanism yet so will think a bit and punt for a later PR.